src/components/Video/Controls/Controls.tsx

```diff
@@ -21,11 +21,38 @@ const Controls = ({
   // Handle play pause
   */
   const [isPlaying, setIsPlaying] = useState(isAutoplaying);
+  const [shareLinkText, setShareLinkText] = useState('Share');
+  const [isShareComponent, setIsShareComponent] = useState(false);
+
+  const hasNavigatorShare = !!navigator.share;
+
+  const handleCopyLink = () => {
+    navigator.clipboard.writeText(window.location.href);
+  };
+
+  const handleShareVia = async () => {
+    navigator.share({
+      title: 'Video',
+      text: 'Video',
+      url: window.location.href,
+    });
+  };
 
   const handlePlayToggle = useCallback(() => {
     setIsPlaying(prev => !prev);
   }, []);
 
+  const handleShareToggle = useCallback(() => {
+    if (!hasNavigatorShare) {
+      setShareLinkText('Copied!');
```

Comment:
I would also add a timeout here, so that the text changes to `Copied!`, but then resets after 5 seconds or something. So that you can copy it again if you need to.